File: ziteng/Maximum Subarray.java
public class Solution {
    public int maxSubArray(int[] A) {
        int newsum=A[0];
        int max=A[0];
        for(int i=1;i<A.length;i++){
            newsum=Math.max(newsum+A[i],A[i]);
            max= Math.max(max, newsum);
        }
        return max;
    }
}

File: mu_wang/9_10/Maximum_Subarray .py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums)<1:
            return 0
        curMax = nums[0]
        result = nums[0]
        for i in range(1,len(nums)):
            curMax = max(nums[i],curMax+nums[i])
            result = max(result,curMax)
            
        return result